crates/extensions/tedge-p11-server/src/pkcs11.rs

```diff
@@ -541,6 +550,23 @@ pub mod uri {
                 other,
             })
         }
+
+        /// Add new attributes from `other` to `self`.
+        ///
+        /// If other contains new attributes not present in self, add them to self. If these
+        /// attributes are already present in self, preserve value currently in self.
+        pub fn append_attributes(&mut self, other: Self) {
+            self.token = self.token.take().or(other.token);
+            self.serial = self.serial.take().or(other.serial);
+            self.id = self.id.take().or(other.id);
+            self.object = self.object.take().or(other.object);
+
+            for (attribute, value) in other.other {
+                if !self.other.contains_key(attribute) {
+                    self.other.insert(attribute, value);
+                }
+            }
```

Comment:
[minor] I had to read this twice because I have been confused by the word `other` used to name two really different things.